lib/modifiers/batch.js

```diff
@@ -0,0 +1,171 @@
+var through = require('through2');
+var _ = require('lodash');
+
+/**
+ * Returns a new stream that batches writes based on a specific interval, count of items, or both.
+ * All items read on the stream and will be re-emitted as part of an array of one or more items
+ * based on the criteria defined in the options.
+ *
+ * @method
+ * @static
+ * @since    1.5.0
+ * @category Util
+ *
+ * @param   {Object} [options]       - Define how items will be batched and written to the
+ *                                     the stream. Note that when both `time` and `count` are
+ *                                     defined, items will be written to the stream whenever
+ *                                     either one of the conditions is met.
+ * @param   {Number} [options.time]  - The interval, in milliseconds, to limit writes to.
+ *                                     During that time, all items read on the stream will
+ *                                     be collected and written as an array at the set
+ *                                     interval.
+ * @param   {Number} [options.count] - The number of items to buffer before writing all
+ *                                     of them in an array.
+ * @returns {Stream}                 - Transform stream.
+ *
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ time: 100 }));
+ *
+ * input.write(1);
+ *
+ * setTimeout(function() {
+ *   input.write(2);
+ *   input.write(3);
+ * }, 100);
+ *
+ * setTimeout(function() {
+ *   input.write(4);
+ *   input.write(5);
+ * }, 200);
+ *
+ * // => [[1], [2, 3], [4, 5]]
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ count: 2 }));
+ *
+ * input.write(1);
+ * input.write(2);
+ * input.write(3);
+ * input.write(4);
+ * input.write(5);
+ *
+ * // => [[1, 2], [3, 4], [5]]
+ */
+function batch(options) {
+  var settings = _.extend({
```

Comment:
Good point. I will add that. I would have assumed this should throw, but now I see that some of the other modifiers (like `pluck`) expect the error to emit on the stream instead. It also looks like `pluck` will only emit an error on the first read it does, meaning that if there is an empty stream, it will allow you to successfully use bad options. Why am I saying all of that again? Oh yeah, how do you want me to handle validation?